### DISCUSSION TITLE
Pipe character inside code

### DISCUSSION THREAD

**Discussion by chenxsan** [remarkjs/remark#573] [Q&A] [answered]
Hi all, I have a problem with remark and any help would be appreciated.

As you might notice that we can use `|` in code as below [on github](https://docs.github.com/en/free-pro-team@latest/github/writing-on-github/organizing-information-with-tables#formatting-content-within-your-table):

```
| First Header | Second Header          |
| ------------ | ---------------------- |
| Content Cell | Content `a\|code` Cell |
| Content Cell | Content Cell           |
```
And it renders into:

| First Header | Second Header          |
| ------------ | ---------------------- |
| Content Cell | Content `a\|code` Cell |
| Content Cell | Content Cell           |

But remark won't remove that `\` before `|` as I found:

```
<td>Content <code>a\|code</code> Cell</td>
```
<table>
<thead>
<tr>
<th>First Header</th>
<th>Second Header</th>
</tr>
</thead>
<tbody>
<tr>
<td>Content Cell</td>
<td>Content <code>a\|code</code> Cell</td>
</tr>
<tr>
<td>Content Cell</td>
<td>Content Cell</td>
</tr>
</tbody>
</table>

As far as I learned so far, `remark@12` has `gfm` option enabled by default, hence I expect it to work the same as Github's behavior. So what can I do to eliminate that `\` before `|`?

I've setup a repository here https://github.com/chenxsan/remark-pipe-in-table, you can clone it and run `yarn` to install then `node run.js` to check the output.

**Comment by wooorm** [comment, marked as answer]
I was under the impressions this was working. But, anyway:

```
yarn add remark-parse remark-gfm -S
```

to update remark-parse and then and using `remark-gfm` in your `run` code solves the issue

**Reply by chenxsan** [reply]
Do you have solution for remark@12? As I'm directly using [remark-loader ](https://github.com/webpack-contrib/remark-loader) and they're running remark@12 at the moment. I'm afraid it will take some time for them to upgrade to remark 13 https://github.com/webpack-contrib/remark-loader/issues/30

**Reply by wooorm** [reply]
as you’re using yarn, maybe resolutions might help? https://classic.yarnpkg.com/en/docs/selective-version-resolutions/ (I don‘t use yarn myself)

**Reply by chenxsan** [reply]
I've tested with the latest `remark-parse` and `remark-gfm`, it worked as expected.

**Reply by wooorm** [reply]
I think this was just never handled in remark-parse. Maybe bugging the webpack folks to get an update relatively soon? Nothing really changes on their side 🤷‍♂️

**Reply by chenxsan** [reply]
Thanks, I'll post more details over there.
